src/redisearch_rs/c_entrypoint/slots_tracker_ffi/cbindgen.toml

```diff
@@ -0,0 +1,68 @@
+includes = ["redismodule.h"]
+sys_includes = ["stdatomic.h"]
+language = "C"
+autogen_warning = "/* Warning, this file is autogenerated by cbindgen from `src/redisearch_rs/c_entrypoint/slots_tracker_ffi/build.rs. Don't modify it manually. */"
+pragma_once = true
+
+after_includes = """
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// static definitions in header                                                                             //
+// We want to avoid atomic read access through FFI boundary, so we maintain atomic version counter here.    //
+// This removes any overhead when checking versions from the query execution path.                          //
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// Global version counter for the key space state.
+// Aligned with the definition in result_processor.c
+extern atomic_uint key_space_version;
+
+/**
+ * Sets the local slot ranges this shard is responsible for.
+ *
+ * This function updates the "local slots" set to match the provided ranges.
+ * If the ranges differ from the current configuration:
+ * - Updates "local slots" to the new ranges
+ * - Removes any overlapping slots from "fully available slots" and "partially available slots"
+ * - Increments the version counter
+ *
+ * If the ranges are identical to the current configuration, no changes are made.
+ *
+ * # Safety
+ *
+ * This function must be called from the main thread only.
+ * The `ranges` pointer must be valid and point to a properly initialized RedisModuleSlotRangeArray.
+ * The ranges array must contain `num_ranges` valid elements.
+ * All ranges must be sorted and have start <= end, with values in [0, 16383].
+ */
+static void slots_tracker_set_local_slots(const RedisModuleSlotRangeArray *ranges) {
+  uint32_t version_before = atomic_load_explicit(&key_space_version, memory_order_relaxed);
+  uint32_t version_after = slots_tracker_set_local_slots_internal(ranges);
+  if (version_after != version_before) {
+    atomic_store_explicit(&key_space_version, version_after, memory_order_relaxed);
+  }
+}
```

Comment:
I assume this means you can't define this function in the Rust FFI layer?

Reply:
@GuyAv46 ?